enterprise/frontend/src/embedding-sdk/cli/steps/setup-permission.ts

```diff
@@ -0,0 +1,207 @@
+import { search } from "@inquirer/prompts";
+import chalk from "chalk";
+import toggle from "inquirer-toggle";
+
+import { SANDBOXED_GROUP_NAMES } from "../constants/config";
+import { NOT_ENOUGH_TENANCY_COLUMN_ROWS } from "../constants/messages";
+import type { CliStepMethod } from "../types/cli";
+import { getCollectionPermissions } from "../utils/get-collection-permissions";
+import { getPermissionsForGroups } from "../utils/get-permission-groups";
+import { getTenancyIsolationSandboxes } from "../utils/get-tenancy-isolation-sandboxes";
+import { printHelperText } from "../utils/print";
+import {
+  cliError,
+  propagateErrorResponse,
+} from "../utils/propagate-error-response";
+import { sampleTenantIdsFromTables } from "../utils/sample-tenancy-column-values";
+
+export const setupPermissions: CliStepMethod = async state => {
```

Comment:
TODO: this step is getting quite long. I will extract them to separate functions and steps afterwards.

Reply:
if these are in a logical block, then i'd say we can keep this step and extract things into their own functions (but not necessarily CLI steps)